test/test_view_dvh_tab.py

```diff
@@ -83,18 +86,54 @@ def test_object():
     return test
 
 
+def wait_for_widget(layout, timeout=1000):
+    start_time = time.time()
+    while time.time() - start_time < timeout/1000:
+        if layout and layout.count() > 1:
+            return True
+        QApplication.processEvents()
+        time.sleep(0.1)
+    return False
+
+
 def test_dvh_tab_with_dvh_not_calculated(qtbot, test_object, init_config):
     if test_object.main_window.dvh_tab.patient_dict_container.has_attribute("raw_dvh"):
         test_object.main_window.dvh_tab.patient_dict_container.additional_data.pop("raw_dvh")
     assert test_object.main_window.dvh_tab.patient_dict_container.has_attribute("raw_dvh") is False
     # check that Calculate DVH tab must appear
 
     test_object.main_window.show()
+    layout = test_object.main_window.dvh_tab.dvh_tab_layout.itemAt(1).layout()
```

Comment:
The entire determination of the button_calc_dvh warrants a rewrite with an explanation of how it is being "discovered" and why the specific approach is being used.  Will this fail if there is a change in layout in the application?  Is there a more dynamic way of getting to the "right answer"?
This test was failing on MacOS 14.7 (Sonoma)/Apple Silicon, and it took these changes to get it to pass.

Reply:
Over time, with so many teams, there is much to rethink and refactor.